conda/gateways/connection/session.py

```diff
@@ -60,6 +64,48 @@ def close(self):
         raise NotImplementedError()
 
 
+def get_channel_name_from_url(url: str) -> str | None:
+    """
+    Given a URL, determine the channel it belongs to and return its name.
+    """
+    for channel in get_channel_objs(context):
+        for base_url in channel.base_urls:
+            if url.lower().startswith(base_url):
+                return channel.canonical_name
+
+
+@lru_cache(None)
+def session_manager(url: str):
+    """
+    Function that determines the correct Session object to be returned
+    based on the URL that is passed in.
+    """
+    channel_name = get_channel_name_from_url(url)
+
+    # If for whatever reason a channel name can't be determined, (should be unlikely)
+    # we just return the default session object.
+    if channel_name is None:
+        return CondaSession()
+
+    channel_settings = {}
+    for settings in context.channel_settings:
+        if settings.get("channel") == channel_name:
+            channel_settings = settings
```

Comment:
tests?

Reply:
Nice catch. I just added another test that should bump the coverage of this function to 💯.